### ISSUE TITLE
[Windows 64-bit] 64-bit AIR application crashes with Logitech cameras

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#53]
## Problem Description
64-bit Windows AIR application crashes when trying to use Logitech cameras.
When you just try to get Camera with connected Logitech camera application always crashes.
There is no way to use 64-bit AIR applications with Logitech cameras.
It has been tested with AIR 32.0.0.89 and AIR 32.0.0.103 beta with 64-bit AIR runtime.
Tested with multiple Logitech HD Webcam C270, Logitech C960 and some others.
Same problem with both AIR runtime versions and with multiple Logitech cameras. 
It works fine 32-bit AIR runtime.

Tracker link: https://tracker.adobe.com/#/view/AIR-4198808

## Steps to Reproduce
Launch 64-bit AIR application which try to get connected cameras names and trace it.
```
var cameras:Array = Camera.names;
trace("Cameras:", cameras);
```

**Actual Result:**
Application crashes with:
Faulting application name: camera_64bit_crash.exe, version: 0.0.0.0, time stamp: 0x5c5b5e97
Faulting module name: Adobe AIR.dll, version: 32.0.0.103, time stamp: 0x5c5b61fe
Exception code: 0xc0000409
Fault offset: 0x000000000082c9bc
Faulting process ID: 0x3814
Faulting application start time: 0x01d4d8dee0630e74
Faulting application path: C:\camera_64bit_crash\camera_64bit_crash.exe
Faulting module path: C:\camera_64bit_crash\Adobe AIR\Versions\1.0\Adobe AIR.dll
Report ID: 1ec90cc0-44d2-11e9-83ba-7a31c1cfc3e0
Faulting package full name: 
Faulting package-relative application ID: 

**Expected Result:**
Application not crash and trace cameras names:
Cameras: Logitech HD Webcam C270

## Known Workarounds
none

**Comment by itlancer** [comment]
Application still crashes with latest AIR 33.1.1.190 with crash log:
```
Application crashes with:
Faulting application name: camera_64bit_crash.exe, version: 0.0.0.0, time stamp: 0x5f075b01
Faulting module name: Adobe AIR.dll, version: 33.1.1.190, time stamp: 0x5f0763c9
Exception code: 0xc0000409
Fault offset: 0x000000000083d77c
Faulting process ID: 0x614
Faulting application start time: 0x01d65e6c2846dcf2
Faulting application path: C:\camera_64bit_crash\camera_64bit_crash.exe
Faulting module path: C:\camera_64bit_crash\Adobe AIR\Versions\1.0\Adobe AIR.dll
Report ID: f9bee45f-494d-4710-a545-47a6a85e5a2f
Faulting package full name:
Faulting package-relative application ID:
```

**Comment by ajwfrost** [comment]
Thanks - is there any chance we can get a dump file of this? That fault offset is within the libvcruntime so it's in a call that we're making, but we could do with seeing the call stack for this! 
https://helgeklein.com/blog/2018/10/creating-an-application-crash-dump/

**Comment by itlancer** [comment]
@ajwfrost, of course.
Here dump: [camera_64bit_crash.exe.17324.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4952585/camera_64bit_crash.exe.17324.zip)

**Comment by NicolasDionB** [comment]
Any update on this? I'm still having the issue but specifically when LogiCapture is installed on my computer. If I uninstall it, the crashes stop. I'm built my app using the latest SDK, 33.1.1.300 even if there was a fix in 33.1.1.176: _Gamua-320: Fixing crash when enumerating cameras with Logitech driver_.

**Comment by NicolasDionB** [comment]
Actually, the fix is supposed to be for 32bits AIR app but I get the opposite result. For me, the app crashes **if I do not specify** `<architecture>64</architecture>` in my application.xml file. I can confirm that in my task manager, the app is really in 64 bits and it works!! When I remove the tag, publish and run again, the app is marked as 32 bits and it crashes on camera access. So for me, the fix was for 64 bits and not 32 bits. Am I the only one or is the fix tagged on the wrong issue (#320 instead of this one)?

**Comment by itlancer** [comment]
For me it still crashes only with 64-bit AIR applications with latest AIR SDK 33.1.1.300 (when Logitech camera plugged in) with crash log:
```
Application crashes with: camera_64bit_crash.exe, version: 0.0.0.0, time stamp: 0x5f9774d6
Faulting module name: Adobe AIR.dll, version: 33.1.1.300, time stamp: 0x5f977bc6
Exception code: 0xc0000409
Fault offset: 0x000000000083d65c
Faulting process ID: 0x463c
Faulting application start time: 0x01d6bf428f5aea3e
Faulting application path: C:\camera_64bit_crash\camera_64bit_crash.exe
Faulting module path: C:\camera_64bit_crash\Adobe AIR\Versions\1.0\Adobe AIR.dll
Report ID: f94bc03e-bb29-4470-b773-3181de99d740
Faulting package full name: 
Faulting package-relative application ID: 
```

32-bit AIR applications not crashes with my tests with latest AIR SDK 33.1.1.300 (after 33.1.1.176 fix which was fixed with https://github.com/Gamua/Adobe-Runtime-Support/issues/320).
So current original issue still not fixed for me.

**Comment by NicolasDionB** [comment]
Any update on this?

**Comment by itlancer** [comment]
For me it still crashes with latest AIR 33.1.1.476.
Plug in Logitech C270 camera (it will use lvcod64.dll driver file), compile attached example (for Windows 64-bit) and go through described steps.
You will get crash:
```
Application crashes with: camera_64bit_crash.exe, version: 0.0.0.0, time stamp: 0x608ac525
Faulting module name: Adobe AIR.dll, version: 33.1.1.476, time stamp: 0x608ace13
Exception code: 0xc0000409
Fault offset: 0x000000000078e16c
Faulting process ID: 0x336c
Faulting application start time: 0x01d74b257068c108
Faulting application path: C:\camera_64bit_crash\camera_64bit_crash.exe
Faulting module path: C:\camera_64bit_crash\Adobe AIR\Versions\1.0\Adobe AIR.dll
Report ID: e6040bdb-c921-4d2e-86a1-20e44a6611a1
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks
